fileshare_process/grpc_fileshare_process.go

```diff
@@ -0,0 +1,175 @@
+package fileshare_process
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/NordSecurity/nordvpn-linux/fileshare/pb"
+	"github.com/NordSecurity/nordvpn-linux/internal"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/status"
+)
+
+type StartupErrorCode int
+
+const (
+	CodeAlreadyRunning StartupErrorCode = iota + 1
+	CodeAlreadyRunningForOtherUser
+	CodeFailedToCreateUnixScoket
+	CodeMeshnetNotEnabled
+	CodeAddressAlreadyInUse
+	CodeFailedToEnable
+)
+
+var (
+	ErrAlreadyRunning             = errors.New("already running")
+	ErrAlreadyRunningForOtherUser = errors.New("already running for other user")
+	ErrFailedToCreateUnixScoket   = errors.New("failed to create unix socket")
+	ErrMeshnetNotEnabled          = errors.New("meshnet not enabled")
+	ErrAddressAlreadyInUse        = errors.New("address already in use")
+	ErrFailedToEnable             = errors.New("failed to enable")
+)
+
+var errorCodeToError = map[StartupErrorCode]error{
+	CodeAlreadyRunning:             ErrAlreadyRunning,
+	CodeAlreadyRunningForOtherUser: ErrAlreadyRunningForOtherUser,
+	CodeFailedToCreateUnixScoket:   ErrFailedToCreateUnixScoket,
+	CodeMeshnetNotEnabled:          ErrMeshnetNotEnabled,
+	CodeAddressAlreadyInUse:        ErrAddressAlreadyInUse,
+	CodeFailedToEnable:             ErrFailedToEnable,
+}
+
+const (
+	FileshareSocket     = "/tmp/fileshare.sock"
```

Comment:
Fileshare socket is not a permanent file, so `/tmp` seems like a good place to store it. Less of a hassle with managing the socket file for us. This will be snaps contained `/tmp`, not the hosts `/tmp`.